packages/wallet-adapter-core/src/WalletCore.ts

```diff
@@ -239,44 +244,41 @@ export class WalletCore extends EventEmitter<WalletCoreEvents> {
   }
 
   /**
-   * Set potential Standard compatible SDK + extension wallets
-   *
-   * Loop over local SDK and Extensions wallets
-   * 1) check it is Standard compatible
-   * 2) Update their readyState to Installed (for a future UI detection)
-   * 3) push the wallet into a local wallets array
-   * 4) standardize each wallet to the Wallet Plugin type interface for legacy compatibility
+   * Set AIP-62 SDK wallets
+   */
+  private fetchSDKAIP62AptosWallets() {
+    this._sdkWallets.map((wallet: AptosStandardWallet) => {
+      this.standardizeAIP62WalletType(wallet);
+    });
+  }
+
+  /**
+   * Set AIP-62 extension wallets
    *
    * @param extensionwWallets
    */
-  private setWallets(extensionwWallets: readonly AptosWallet[]) {
-    const aptosStandardWallets: AptosStandardWallet[] = [];
-
-    [...this._sdkWallets, ...extensionwWallets].map(
-      (wallet: AptosStandardWallet) => {
-        if (this.excludeWallet(wallet.name)) {
-          return;
-        }
-        const isValid = isWalletWithRequiredFeatureSet(wallet);
-        if (isValid) {
-          wallet.readyState = WalletReadyState.Installed;
-          aptosStandardWallets.push(wallet);
-          this.standardizeStandardWalletToPluginWalletType(wallet);
-        }
-      }
+  private setExtensionAIP62Wallets(extensionwWallets: readonly AptosWallet[]) {
+    // Twallet SDK fires a register event so the adapter assumes it is an extension wallet
+    // so filter out t wallet, remove it when twallet fixes it
+    const wallets = extensionwWallets.filter(
+      (wallet) => wallet.name !== "Dev T wallet" && wallet.name !== "T wallet"
     );
```

Comment:
The issue with Twallet here, is that they fire a register event which the dapp listens to, but since the adapter already includes the wallet package, the adapter "thinks" there are 2 Twallets - once coming from the npm dependency and one coming from the register event.

A npm/sdk wallet, that is already installed and included in WalletCore doesnt need to fire any event to tell the browser "hi i am here" as it is already installed as a dependency.